### ISSUE TITLE
Seeing "invalid TOML syntax" for valid TOML

### ISSUE THREAD

**Issue by lorenmh** [influxdata/telegraf#10534]
### Relevent telegraf.conf

```toml
my_array = [
  # hello I am a comment
  "hello",
]
```

### Logs from Telegraf

invalid TOML syntax

### System info

master branch telegraf

### Steps to reproduce

check out my fork here: https://github.com/lorenmh/telegraf
run `go test ./config/...`

diff here: https://github.com/lorenmh/telegraf/commit/7e44c12cad6e86aeab4036f8f3ed10a6eb7a79f0

### Expected behavior

This is valid TOML and it should be parsed correctly

### Actual behavior

Parse fails, Telegraf won't run

**Comment by powersj** [comment]
I consider BurntSushi/toml the canonical source, this does appear to work and it is valid TOML syntax:

```go
package main

import (
	"fmt"
	"log"

	"github.com/BurntSushi/toml"
)

var tomlData = `
array = [
  # hello I am a comment
  "hello",
  "world"
]
`

type tomlConfig struct {
	Array []string `toml:"array"`
}

func main() {
	var conf tomlConfig
	if _, err := toml.Decode(tomlData, &conf); err != nil {
		log.Fatal(err)
	}

	fmt.Println(conf)
}
```

```s
$  go build && ./test 
{[hello world]}
```

However, we use the https://github.com/influxdata/toml library to parse TOML files. This is really a fork of https://github.com/naoina/toml. I am able to reproduce the invalid TOML syntax with the last release of both libraries:

```
❯ go mod tidy
go: finding module for package github.com/influxdata/toml
go: downloading github.com/influxdata/toml v0.0.0-20180607005434-2a2e3012f7cf
go: found github.com/influxdata/toml in github.com/influxdata/toml v0.0.0-20180607005434-2a2e3012f7cf
go: finding module for package github.com/naoina/go-stringutil
go: found github.com/naoina/go-stringutil in github.com/naoina/go-stringutil v0.1.0
~/test via 🐹 v1.17.6 
❯ go build && ./test 
2022/01/27 12:49:07 toml: line 3: parse error
~/test via 🐹 v1.17.6 
❯ go mod tidy
go: finding module for package github.com/naoina/toml
go: downloading github.com/naoina/toml v0.1.1
go: found github.com/naoina/toml in github.com/naoina/toml v0.1.1
go: finding module for package github.com/kylelemons/godebug/pretty
go: finding module for package github.com/kylelemons/godebug/diff
go: found github.com/kylelemons/godebug/diff in github.com/kylelemons/godebug v1.1.0
go: found github.com/kylelemons/godebug/pretty in github.com/kylelemons/godebug v1.1.0
~/test via 🐹 v1.17.6 
❯ go build && ./test 
2022/01/27 12:52:12 line 3: invalid TOML syntax
```

However, once I upgrade the library to master it is able to parse the TOML:

```
❯ go get github.com/naoina/toml@master
go: downloading github.com/naoina/toml v0.1.2-0.20210730182554-e80af6068b28
go get: upgraded github.com/naoina/toml v0.1.1 => v0.1.2-0.20210730182554-e80af6068b28
go build && ./test 

array = [
  # hello I am a comment
  "hello",
  "world"
]
```

It looks like we are missing https://github.com/naoina/toml/commit/605f28746f043884cd39260015ccf18005f75f4c this commit for comments in arrays.

Next steps: look to update our fork with the missing commit or commits from the https://github.com/naoina/toml library

**Comment by sspaink** [comment]
Looks like this is a duplicate of: https://github.com/influxdata/telegraf/issues/3642, I went ahead and closed #3642 in favor of this one because I think this issue has more information to help resolve the problem.

It would be nice if we could switch to  BurntSushi/toml, from the comments in #3642 and https://github.com/influxdata/telegraf/issues/6474 it seems this was the intention at some point.
